pulsar-client-cpp/lib/ConsumerImpl.h

```diff
@@ -135,6 +137,7 @@ enum ConsumerTopicType {
     ExecutorServicePtr listenerExecutor_;
     ConsumerTopicType consumerTopicType_;
     UnboundedBlockingQueue<Message> incomingMessages_;
+    UnboundedBlockingQueue<ReceiveCallback> pendingReceives_;
```

Comment:
Sure. actually, my motivation to use `UnboundedBlockingQueue` to have thread-safe, unbounded collection. Do you see any specific issue with `UnboundedBlockingQueue`?